server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/CassandraRabbitMQJamesServerMain.java

```diff
@@ -20,24 +20,54 @@
 package org.apache.james;
 
 import static org.apache.james.CassandraJamesServerMain.REQUIRE_TASK_MANAGER_MODULE;
+import static org.apache.james.modules.blobstore.BlobStoreChoosingConfiguration.readBlobStoreChoosingConfiguration;
 
+import org.apache.commons.configuration2.ex.ConfigurationException;
 import org.apache.james.modules.DistributedTaskManagerModule;
 import org.apache.james.modules.TaskSerializationModule;
-import org.apache.james.modules.blobstore.BlobStoreChoosingModule;
+import org.apache.james.modules.blobstore.BlobStoreCacheConfiguredModulesSupplier;
+import org.apache.james.modules.blobstore.BlobStoreChoosingConfiguration;
+import org.apache.james.modules.blobstore.ChoosingBlobStoreConfiguredModulesSupplier;
 import org.apache.james.modules.event.RabbitMQEventBusModule;
 import org.apache.james.modules.rabbitmq.RabbitMQModule;
 import org.apache.james.modules.server.JMXServerModule;
+import org.apache.james.server.core.configuration.Configuration;
+import org.apache.james.server.core.filesystem.FileSystemImpl;
+import org.apache.james.utils.PropertiesProvider;
 
+import com.google.common.collect.ImmutableList;
 import com.google.inject.Module;
 import com.google.inject.util.Modules;
 
 public class CassandraRabbitMQJamesServerMain implements JamesServerMain {
     public static final Module MODULES =
         Modules
             .override(Modules.combine(REQUIRE_TASK_MANAGER_MODULE, new DistributedTaskManagerModule()))
-            .with(new RabbitMQModule(), new BlobStoreChoosingModule(), new RabbitMQEventBusModule(), new TaskSerializationModule());
+            .with(new RabbitMQModule(), new RabbitMQEventBusModule(), new TaskSerializationModule());
 
     public static void main(String[] args) throws Exception {
-        JamesServerMain.main(MODULES, new JMXServerModule());
+        Configuration configuration = Configuration.builder()
+            .useWorkingDirectoryEnvProperty()
+            .build();
+
+        BlobStoreChoosingConfiguration blobStoreChoosingConfiguration = parseBlobStoreChoosingConfiguration(configuration);
+
+        Module baseModule = baseModule(blobStoreChoosingConfiguration);
+
+        JamesServerMain.main(configuration,
+            ImmutableList.of(baseModule, new JMXServerModule()));
+    }
+
+    static BlobStoreChoosingConfiguration parseBlobStoreChoosingConfiguration(Configuration configuration) throws ConfigurationException {
+        PropertiesProvider propertiesProvider = new PropertiesProvider(new FileSystemImpl(configuration.directories()), configuration);
+        return readBlobStoreChoosingConfiguration(propertiesProvider);
+    }
+
+    public static Module baseModule(BlobStoreChoosingConfiguration blobStoreChoosingConfiguration) {
+        return Modules.combine(ImmutableList.<Module>builder()
+                .add(MODULES)
```

Comment:
it would inline the MODULES definition here to avoid spreading modules far from each others

Reply:
Shared with the LDAP version that's not doable without code duplication.

If I do duplicate we take the risk of 'forgetting modifying the LDAP version.

Proposition: handle LDAP choice with a configuration POJO and not a separate product? Tell me if I should modify the ADR accordingly.

Reply:
I think it's a solution that @rouazana would like.
I don't really have opinion regarding that

Reply:
Yes, I still see no value having a separated product for LDAP version (vs local repository) while it can be configured and benefit each product.